DOCS.md

```diff
@@ -8,7 +8,143 @@ Registry: https://hub.docker.com/r/target/vela-makisu
 
 ## Usage
 
-COMING SOON!
+_Notes:_
+
+* The plugin supports reading all parameters via environment variables or files. Values set as a file take precedence over default values set from the environment.
+* We do not recommended using latest for pipelines. Users should use pinned images to decrease volatility of external changes to their pipelines. 
+
+Sample of building and publishing an image:
+
+```yaml
+steps:
+  - name: publish_hello-world
+    image: target/vela-makisu:latest
+    parameters:
+      registry: index.docker.io
+      tag: index.docker.io/octocat/hello-world
+      pushes: [ index.docker.io ]
+```
+
+Sample of building an image without publishing:
+
+```diff
+steps:
+  - name: publish hello world
+    image: target/vela-makisu:latest
+    parameters:
+-     pushes: [ index.docker.io ]
+      registry: index.docker.io
+      tag: index.docker.io/octocat/hello-world:latest
+```
+
+Sample of building and publishing an image with custom tags:
+
+```diff
+steps:
+  - name: publish hello world
+    image: target/vela-makisu:latest
+    parameters:
+      registry: index.docker.io
+      tag: index.docker.io/octocat/hello-world:latest
+      pushes: [ index.docker.io ]
++     replicas:
++       - index.docker.io/octocat/hello-world:1
++       - index.docker.io/octocat/hello-world:foobar
+```
+
+Sample of building and publishing an image with build arguments:
+
+```diff
+steps:
+  - name: publish hello world
+    image: target/vela-makisu:latest
+    pull: true
```

Comment:
I noticed that some of the steps have no `pull` attribute and some do.

Do you think we could update this one way or the other to keep them consistent?

Reply:
Meant to remove all of them. WIll update